rewrite-java/src/main/java/org/openrewrite/java/JavaVisitor.java

```diff
@@ -423,7 +423,8 @@ public J visitCase(J.Case case_, P p) {
         } else {
             c = (J.Case) temp;
         }
-        c = c.getPadding().withExpressions(visitContainer(c.getPadding().getExpressions(), JContainer.Location.CASE_EXPRESSION, p));
+        c = c.getPadding().withCaseLabels(visitContainer(c.getPadding().getCaseLabels(), JContainer.Location.CASE_LABEL, p));
+        c = c.withGuard((Expression) visitExpression(c.getGuard(), p));
```

Comment:
Please note that `visitExpression()` would be wrong to call here. This is a very special visitor method, which is only there to allow subclasses to implement a visit method that ends up getting called for every LST type implementing the `Expression` interface. By default it does nothing.

```suggestion
        c = c.withGuard(visitAndCast(c.getGuard(), p));
```